types/src/gas.rs

```diff
@@ -1,34 +1,42 @@
-use std::{fmt, iter::Sum};
+//! The `gas` module is used for working with Gas including converting to and from Motes.
 
-use num::Zero;
+use core::{fmt, iter::Sum};
 
-use casper_types::U512;
+use num::Zero;
 
-use crate::shared::motes::Motes;
+use crate::{Motes, U512};
 
+/// The `Gas` struct represents a `U512` amount of gas.
 #[derive(Debug, Default, Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
 pub struct Gas(U512);
 
 impl Gas {
+    /// Constructs a new `Gas`.
     pub fn new(value: U512) -> Self {
         Gas(value)
     }
 
+    /// Returns the inner `U512` value.
     pub fn value(&self) -> U512 {
         self.0
     }
 
+    /// Returns an `Option<Gas>`. Returns `None` if the conversion cannot be performed.
     pub fn from_motes(motes: Motes, conv_rate: u64) -> Option<Self> {
         motes
             .value()
             .checked_div(U512::from(conv_rate))
             .map(Self::new)
     }
 
+    /// Safe addition method that returns an `Option<Gas>`. Returns `None` if an overflow has
+    /// occurred.
```

Comment:
This is taken from the std lib docs:
```suggestion
    /// Checked integer addition. Computes `self + rhs`, returning `None` if overflow occurred.
```